pkg/pillar/cmd/zedmanager/handlevolumemgr.go

```diff
@@ -6,173 +6,171 @@ package zedmanager
-		base.OldVolumeConfigLogType, key).Infof("App instance to volume relation.")
-	log.Infof("AddOrRefcountVolumeConfig done for %s", key)
+		base.VolumeRefConfigLogType, key).Infof("App instance to volume relation.")
+	log.Infof("MaybeAddVolumeRefConfig done for %s", key)
 }
 
-// MaybeRemoveVolumeConfig decreases the RefCount and deletes the VolumeConfig
+// MaybeRemoveVolumeRefConfig decreases the RefCount and deletes the VolumeRefConfig
 // when the RefCount reaches zero
-func MaybeRemoveVolumeConfig(ctx *zedmanagerContext, blobSha256 string,
-	appInstID uuid.UUID, volumeID uuid.UUID, purgeCounter uint32) {
+func MaybeRemoveVolumeRefConfig(ctx *zedmanagerContext, appInstID uuid.UUID,
+	vrs types.VolumeRefStatus) {
 
-	key := types.VolumeKeyFromParts(blobSha256, appInstID, volumeID, purgeCounter)
-	log.Infof("MaybeRemoveVolumeConfig for %s", key)
-	m := lookupVolumeConfig(ctx, key)
+	log.Infof("MaybeRemoveVolumeRefConfig for %s", vrs.Key())
+	m := lookupVolumeRefConfig(ctx, vrs.Key())
 	if m == nil {
-		log.Infof("MaybeRemoveVolumeConfig: config missing for %s", key)
+		log.Infof("MaybeRemoveVolumeRefConfig: config missing for %s", vrs.Key())
 		return
 	}
 	if m.RefCount == 0 {
-		log.Fatalf("MaybeRemoveVolumeConfig: Attempting to reduce "+
-			"0 RefCount for %s", key)
+		log.Fatalf("MaybeRemoveVolumeRefConfig: Attempting to reduce "+
+			"0 RefCount for %s", vrs.Key())
 	}
 	m.RefCount--
 	if m.RefCount == 0 {
-		log.Infof("MaybeRemoveVolumeConfig deleting %s", key)
-		unpublishVolumeConfig(ctx, key)
+		log.Infof("MaybeRemoveVolumeRefConfig deleting %s", vrs.Key())
+		unpublishVolumeRefConfig(ctx, vrs.Key())
 	} else {
-		log.Infof("MaybeRemoveVolumeConfig remaining RefCount %d for %s",
-			m.RefCount, key)
-		publishVolumeConfig(ctx, m)
+		log.Infof("MaybeRemoveVolumeRefConfig remaining RefCount %d for %s",
+			m.RefCount, vrs.Key())
+		publishVolumeRefConfig(ctx, m)
 	}
 	base.NewRelationObject(base.DeleteRelationType, base.AppInstanceConfigLogType, appInstID.String(),
-		base.OldVolumeConfigLogType, key).Infof("App instance to volume relation.")
+		base.VolumeRefConfigLogType, vrs.Key()).Infof("App instance to volume relation.")
 }
 
-func lookupVolumeConfig(ctx *zedmanagerContext, key string) *types.OldVolumeConfig {
+func lookupVolumeRefConfig(ctx *zedmanagerContext, key string) *types.VolumeRefConfig {
 
-	pub := ctx.pubVolumeConfig
+	pub := ctx.pubVolumeRefConfig
 	c, _ := pub.Get(key)
 	if c == nil {
-		log.Infof("lookupVolumeConfig(%s) not found", key)
+		log.Infof("lookupVolumeRefConfig(%s) not found", key)
 		return nil
 	}
-	config := c.(types.OldVolumeConfig)
+	config := c.(types.VolumeRefConfig)
 	return &config
 }
 
-// Note that this function returns the entry even if Pending* is set.
-func lookupVolumeStatus(ctx *zedmanagerContext, blobSha256 string,
-	appInstID uuid.UUID, volumeID uuid.UUID, purgeCounter uint32) *types.OldVolumeStatus {
+func lookupVolumeRefStatus(status *types.AppInstanceStatus,
```

Comment:
This name is confusion; I'd expect it to do the same as lookupVolumeRefConfig i.e., a Get(key).
But see comment about app instance status having an array of VolumeRefStatus separately from the pubsub maintained VolumeRefStatus collection. That is problematic.